packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -1227,66 +1261,6 @@ if (y) {
         }
       `,
     },
-    {
-      options: [{ allowNullableEnum: false }],
-      code: `
-        enum ExampleEnum {
-          This = 'one',
-          That = 'two',
-        }
-        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
-        if (!theEnum) {
-        }
-      `,
-      errors: [
-        {
-          line: 7,
-          column: 14,
-          messageId: 'conditionErrorNullableEnum',
-          endLine: 7,
-          endColumn: 21,
-        },
-      ],
-      output: `
-        enum ExampleEnum {
-          This = 'one',
-          That = 'two',
-        }
-        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
-        if (theEnum == null) {
-        }
-      `,
-    },
-    {
-      options: [{ allowNullableEnum: false }],
-      code: `
-        enum ExampleEnum {
-          This = 1,
-          That = 2,
-        }
-        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
-        if (!theEnum) {
-        }
-      `,
-      errors: [
-        {
-          line: 7,
-          column: 14,
-          messageId: 'conditionErrorNullableEnum',
-          endLine: 7,
-          endColumn: 21,
-        },
-      ],
-      output: `
-        enum ExampleEnum {
-          This = 1,
-          That = 2,
-        }
-        const theEnum = Math.random() < 0.3 ? ExampleEnum.This : null;
-        if (theEnum == null) {
-        }
-      `,
-    },
```

Comment:
This is no longer applicable due to `is('nullish', 'truthy number', 'enum')` and `is('nullish', 'truthy string', 'enum')` conditions being treated as an known edge case for `allowEnum === true`